### ISSUE TITLE
BackgroundPlotter.update_scalars does not work with UnstructuredGrid when smooth_shading=True

### ISSUE THREAD

**Issue by truenicoco** [pyvista/pyvista#668]
Updating scalars on a `PolyData` works fine with `smooth_shading` enabled or disabled. However, when plotting an `UnstructuredGrid`, for example a tetrahedral mesh, updating scalars does not do anything if `smooth_shading` is enabled. From what I could understand, it is because the actual volumetric mesh is not plotted when using `smooth_shading`, just its extracted surface.

**To Reproduce**

```py
import pyvista as pv
import numpy as np
import time
from tetgen import TetGen

surf_mesh = pv.Sphere()

tet = TetGen(surf_mesh)
tet.make_manifold()
tet.tetrahedralize()
mesh = tet.grid
            
mesh.point_arrays['x'] = np.arange(len(mesh.points))

for smooth in False, True:
    p = pv.BackgroundPlotter()
    p.add_mesh(mesh, smooth_shading=smooth)
    time.sleep(1)
    p.update_scalars(-np.arange(len(mesh.points)), mesh)
    time.sleep(1)
    p.close()

```
-----

**System Information:**

 I am not sure this is relevant, but since it was in the issue template…

```
--------------------------------------------------------------------------------
  Date: Tue Apr 07 17:44:38 2020 CEST

                OS : Linux
            CPU(s) : 4
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : Intel Open Source Technology Center
      GPU Renderer : Mesa DRI Intel(R) HD Graphics 5500 (Broadwell GT2)
       GPU Version : 4.5 (Core Profile) Mesa 19.2.8

  Python 3.7.6 | packaged by conda-forge | (default, Mar  5 2020, 15:27:18)
  [GCC 7.3.0]

           pyvista : 0.24.0
               vtk : 8.2.0
             numpy : 1.18.1
           imageio : 2.8.0
           appdirs : 1.4.3
            scooby : 0.5.2
        matplotlib : 3.2.1
             PyQt5 : 5.12.3
           IPython : 7.13.0
          colorcet : 1.0.0
             scipy : 1.4.1

  Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for
  Intel(R) 64 architecture applications
--------------------------------------------------------------------------------
```

**Comment by banesullivan** [comment]
Yeah, this is a known issue (though not documented anywhere). The `smooth_shading` flag makes a copy of any non-PolyData mesh so you can't reflect changes to the original mesh.

Relevant source:

https://github.com/pyvista/pyvista/blob/127ec0a79bc653bdc5a0deba501d40b2d8f23fac/pyvista/plotting/plotting.py#L1330-L1340

**Comment by truenicoco** [comment]
What would be a "clean" fix for this? Store `ind` as a attribute of the plotter and modify `update_scalars()` accordingly?

**Comment by banesullivan** [comment]
There might be a bit more overhead required as `update_scalars` is just one of several methods that would need to be changed.

I'll look into this to see if there are ways to do this without extracting only the outer surface.

**Comment by truenicoco** [comment]
I don't know if it is worth opening another ticket for this, but I noticed smooth shading also disable the ability to use a clip plane interactively.

**Comment by banesullivan** [comment]
Yeah... that's the result of this same issue, so we can keep it here

**Comment by banesullivan** [comment]
So here is a thought: We could do a rather heavy refactor of the `add_mesh` and `add_volume` methods (needed anyway... see #299) to implement those methods in a pipeline model. That would allow the user to do smooth shading on non-PolyData meshes and easily make changes to the rendered meshes after applying other filters in the rendering pipeline.

Changing to a pipeline model in these methods would make a lot of other things more robust.

Thoughts, @akaszynski?

**Comment by akaszynski** [comment]
Pipeline model?  Like how VTK does it?

**Comment by banesullivan** [comment]
Correct. As in keeping a reference to the data passed in the `mesh` argument then setting input connections in a pipeline like a true VTK workflow. Then would could "inject" routines into the pipeline such as surface extraction and normal calculating filters so that when a user modifies the original input mesh, it will propagate through the pipeline to the renderer.

Do note that those updates could add lag for "larger" datasets but this is exactly how ParaView does it.

Implementing this alone wouldn't be too much (I think), but those methods all need to be refactored anyways.

**Comment by banesullivan** [comment]
It is still the same underlying issue so we can keep this as the place to track it. Perhaps I should rename the title
